### PR TITLE
Correcting error message when dynamic data doen't have any data

### PR DESCRIPTION
**Pull request by jayaranigarg**
#443

### CODE REVIEW

File: src/TestFramework/MSTest.Core/Attributes/DynamicDataAttribute.cs [hunk 1]
```diff
@@ -128,7 +128,8 @@ public IEnumerable<object[]> GetData(MethodInfo methodInfo)
             }
 
             IEnumerable<object[]> enumerable = obj as IEnumerable<object[]>;
-            if (enumerable == null)
+
+            if (enumerable == null || !enumerable.Any())
```
Comment (singhsarab) [inline src/TestFramework/MSTest.Core/Attributes/DynamicDataAttribute.cs]: It should be ArgumentException and not ArgumentNullException in this new case.

File: test/UnitTests/MSTest.Core.Unit.Tests/Attributes/DynamicDataAttributeTests.cs [hunk 2]
```diff
@@ -293,7 +306,18 @@ public static IEnumerable<object[]> NullProperty
...
+            get
+            {
+                return new object[][] { };
+            }
+        }
+
+        /// <summary>
+        /// Gets the wrong test data property.
         /// </summary>
         public static IEnumerable<object[]> WrongDataTypeProperty
```
Comment (singhsarab) [inline test/UnitTests/MSTest.Core.Unit.Tests/Attributes/DynamicDataAttributeTests.cs]: Out of the scope of this PR maybe. Why do we WrongDataTypeProperty? How is that different from Null property.

Comment (jayaranigarg) [reply test/UnitTests/MSTest.Core.Unit.Tests/Attributes/DynamicDataAttributeTests.cs]: I think the intent was different but it is not very clear. Correcting it.

**Comment by jamers99** [comment]
When will this be release, or what version is this fix in?

**Comment by jayaranigarg** [comment]
@jamers99 : version 2.0.0-beta4 has this fix. Checkout the release notes : https://github.com/microsoft/testfx/releases/tag/v2.0.0-beta4

**Pull request merged**